quint/src/names/base.ts

```diff
@@ -0,0 +1,188 @@
+/* ----------------------------------------------------------------------------------
+ * Copyright (c) Informal Systems 2022-2023. All rights reserved.
+ * Licensed under the Apache 2.0.
+ * See License.txt in the project root for license information.
+ * --------------------------------------------------------------------------------- */
+
+/**
+ * Type definitions and utilities for Quint name resolution.
+ *
+ * @author Gabriela Moreira
+ *
+ * @module
+ */
+
+import { QuintType } from '../quintTypes'
+
+/**
+ * Possible kinds for definitions
+ */
+export type DefinitionKind = 'module' | 'def' | 'val' | 'assumption' | 'param' | 'var' | 'const' | 'type'
+
+/**
+ * A minimal representation of a QuintDef, to be stored in `LookupTable` and
+ * `DefinitionsByName`.
+ */
+export interface Definition {
+  kind: DefinitionKind
+  reference: bigint
```

Comment:
```suggestion
  originId: bigint
```

I was always puzzled by this name. The problem with references is that they may be transitive. I think we are always pointing to the original definition, right?

Reply:
This got me thinking (because `originId` seems weird to me here), and actually I think this should be just `id`. I'm sort of using `id` as primary keys an `reference` as foreign keys across the code base, but I think this `Definition` is just a projection of some fields from a `QuintDef`, so it's not like the `QuintDef` is the `origin` here, they are the same thing, only with different representations.

I'll change this to `id`.

Reply:
`id` may be confusing, as we are typically using `id` to denote the unique identifier of the object, that is, the id of `Definition` in this case.

Reply:
whops, this comment was meant to be on this thread: https://github.com/informalsystems/quint/pull/1011#discussion_r1254368644

Reply:
I think this would be a single-line change in Apalache:

https://github.com/informalsystems/apalache/blob/f6bfe31271b38f07b77600525356f3ba4455219d/tla-io/src/main/scala/at/forsyte/apalache/io/quint/QuintIR.scala#L59